src/domain/Reservation/components/ReservationCalendar.tsx

```diff
@@ -0,0 +1,252 @@
+  number: number;
+}
+
+export const WEEKDAYS = ['S', 'M', 'T', 'W', 'T', 'F', 'S'];
+
+// 이벤트 우선순위 (중요도 순)
+const PRIORITY_MAP = {
+  완료: 0,
+  승인: 1,
+  예약: 2,
+  거절: 3,
+  취소: 4,
+};
+
+// 샘플 이벤트 데이터
+const sampleEvents: Event[] = [
+  { id: '1', title: '완료', date: '2025-07-14', color: 'green', number: 1 },
+  { id: '2', title: '예약', date: '2025-07-15', color: 'purple', number: 2 },
+  { id: '3', title: '승인', date: '2025-07-15', color: 'blue', number: 3 },
+  { id: '4', title: '거절', date: '2025-07-15', color: 'orange', number: 4 },
+  { id: '5', title: '승인', date: '2025-07-16', color: 'blue', number: 5 },
+  { id: '6', title: '거절', date: '2025-07-16', color: 'orange', number: 6 },
+  { id: '7', title: '취소', date: '2025-07-17', color: 'red', number: 7 },
+  { id: '8', title: '예약', date: '2025-07-17', color: 'purple', number: 8 },
+  { id: '9', title: '완료', date: '2025-07-17', color: 'green', number: 9 },
+  { id: '10', title: '취소', date: '2025-07-18', color: 'red', number: 10 },
+];
+
+export default function ReservationCalendar() {
+  const [currentDate, setCurrentDate] = useState(dayjs()); //현재 시간을 가져옴
+  const [events] = useState<Event[]>(sampleEvents); //이벤트 데이터를 가져옴
+
+  // 이벤트 데이터를 날짜별로 그룹화하여 memoization으로 성능 최적화 (샘플 이벤트에 적용되는 내용이라 수정 실제 API 연동 시 수정 필요)
+  const eventsByDate = useMemo(() => {
+    const eventMap: Record<string, Event[]> = {};
+
+    events.forEach((event) => {
+      if (!eventMap[event.date]) {
+        eventMap[event.date] = [];
+      }
+      eventMap[event.date].push(event);
+    });
+    return eventMap;
+  }, [events]);
+
+  // 핵심: dayjs 객체와 Moment.js와 유사한 API를 사용합니다.
+  const monthStart = currentDate.startOf('month'); // 현재 월의 시작 날짜
+  const monthEnd = currentDate.endOf('month'); // 현재 월의 마지막 날짜
+  const calendarStart = monthStart.startOf('week'); // 현재 월의 첫 번째 주의 시작 날짜
+  const calendarEnd = monthEnd.endOf('week'); // 현재 월의 마지막 주의 마지막 날짜
+
+  // 날짜 배열을 생성하기 위해 반복문 사용 (currentDate 기준으로 한 달의 날짜를 가져옴)
+  const days = useMemo(() => {
+    const daysArr = [];
+    let day = calendarStart;
+    while (day.isSameOrBefore(calendarEnd)) {
+      daysArr.push(day);
+      day = day.add(1, 'day');
+    }
+    return daysArr;
+  }, [calendarStart, calendarEnd]);
+
+  /**
+   * 특정 날짜의 이벤트 목록을 반환합니다
+   * @param date - 조회할 날짜 (dayjs 객체)
+   * @returns 해당 날짜의 이벤트 배열
+   */
+  const getEventsForDate = (date: dayjs.Dayjs) => {
+    const formattedDate = date.format('YYYY-MM-DD');
+    return eventsByDate[formattedDate] || [];
+  };
+
+  /**
+   * 이벤트를 우선순위에 따라 정렬합니다
+   * @param events - 정렬할 이벤트 배열
+   * @returns 우선순위 순으로 정렬된 이벤트 배열 (완료 > 승인 > 예약 > 거절 > 취소)
+   */
+  const sortEventsByPriority = (events: Event[]) => {
+    return events.sort((a, b) => {
+      const priorityA =
+        PRIORITY_MAP[a.title as keyof typeof PRIORITY_MAP] ?? 999;
+      const priorityB =
+        PRIORITY_MAP[b.title as keyof typeof PRIORITY_MAP] ?? 999;
+      return priorityA - priorityB;
+    });
+  };
```

Comment:
`sortEventsByPriorit`y 함수에서 `events.sort()`는 원본 배열을 **직접 수정(mutate)**합니다. 이 함수는 `days.map` 내부에서 `dayEvents`를 정렬하는 데 사용되는데, `dayEvents`는 `eventsByDate`에 저장된 배열의 참조이므로, `eventsByDate`에 저장된 원본 데이터가 정렬되어 버립니다.

한번 정렬된 날짜의 이벤트는 계속 정렬된 상태로 유지될 거 같은데... 예상치 못한 동작을 유발할 수 있을 거 같습니다. 잠재 버그를 막기 위해 원본을 수정하지 않도록 배열의 복사본을 만들어 정렬하시는 걸 권장드립니다.

```suggestion
  const sortEventsByPriority = (events: Event[]) => {
    return [...events].sort((a, b) => {
      const priorityA =
        PRIORITY_MAP[a.title as keyof typeof PRIORITY_MAP] ?? 999;
      const priorityB =
        PRIORITY_MAP[b.title as keyof typeof PRIORITY_MAP] ?? 999;
      return priorityA - priorityB;
    });
  };
```